File: TelerikUIExamples/TelerikUIExamples/AlertGettingStarted.m
//
//  AlertGettingStarted.m
//  TelerikUIExamples
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

#import <TelerikUI/TelerikUI.h>
#import "AlertGettingStarted.h"
#import "UIButton_Circle.h"

@interface AlertGettingStarted ()
@end

@implementation AlertGettingStarted
{
    UILabel* _textLabel;
}

-(void)viewDidLoad
{
    [super viewDidLoad];
    
    _textLabel = [[UILabel alloc] initWithFrame:CGRectMake(0, 100, CGRectGetWidth(self.view.frame), 44)];
    _textLabel.textAlignment = NSTextAlignmentCenter;
    _textLabel.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleBottomMargin;
    _textLabel.text = @"Please, answer the question!";
    [self.view addSubview:_textLabel];

    [UIButton circleButtonInView:self.view title:@"Answer me" target:self action:@selector(show:)];
}

-(void)show:(id)sender
{
// >> getting-started-alert
    TKAlert *alert = [TKAlert new];
    alert.title = @"Chicken or the egg";
    alert.message = @"Which came first, the chicken or the egg?";
// << getting-started-alert
    
    alert.style.maxWidth = 210;
    alert.dismissMode = TKAlertDismissModeSwipe;
    alert.swipeDismissDirection = TKAlertSwipeDismissDirectionVertical;
    
// >> getting-started-alert-action
    [alert addActionWithTitle:@"Egg" handler:^BOOL (TKAlert *alert, TKAlertAction* action) {
        _textLabel.text = @"It was the egg!";
        return YES;
    }];

    [alert addActionWithTitle:@"Chicken" handler:^BOOL (TKAlert *alert, TKAlertAction* action) {
        _textLabel.text = @"It was the chicken!";
        return YES;
    }];
// << getting-started-alert-action
    
// >> getting-started-alert-show
    [alert show:YES];
// << getting-started-alert-show
}


@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/AlertGettingStarted.swift
//
//  AlertGettingStarted.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

import UIKit

class AlertGettingStarted: TKExamplesExampleViewController {

    let textLabel = UILabel()
    
    override func viewDidLoad() {
        
        super.viewDidLoad()

        textLabel.frame = CGRectMake(0, 100, self.view.frame.size.width, 44);
        textLabel.textAlignment = NSTextAlignment.Center
        textLabel.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleBottomMargin.rawValue)
        textLabel.text = "Please, answer the question?"
        self.view.addSubview(textLabel);
        
        UIButton.circleButtonInView(self.view, title: "Answer me", target: self, action: #selector(AlertGettingStarted.show(_:)))
    }
    
    func show(sender: AnyObject) {
// >> getting-started-alert-swift
        let alert = TKAlert()
        alert.title = "Chicken or the egg?"
        alert.message = "Which came first, the chicken or the egg?"
// << getting-started-alert-swift
        
        alert.style.maxWidth = 210
        alert.dismissMode = TKAlertDismissMode.Swipe
        
// >> getting-started-alert-action-swift
        alert.addActionWithTitle("Egg") { (TKAlert, TKAlertAction) -> Bool in
            self.textLabel.text = "It was the egg"
            return true
        }

        alert.addActionWithTitle("Chicken") { (TKAlert, TKAlertAction) -> Bool in
            self.textLabel.text = "It was the chiken"
            return true
        }
// << getting-started-alert-action-swift
        
// >> getting-started-alert-show-swift
        alert.show(true)
// << getting-started-alert-show-swift
    }
    
}
